commit 933929cd622478bb43afe590670613da2e9ff359
Author: nicm <nicm>
Date:   Fri Nov 20 22:02:54 2015 +0000

    Memory leaks and an uninitialized part of utf8_data, from Patrick Palka.

--- a/cmd-if-shell.c
+++ b/cmd-if-shell.c
@@ -97,6 +97,7 @@
 			cmd = args->argv[1];
 		else if (args->argc == 3)
 			cmd = args->argv[2];
+		free(shellcmd);
 		if (cmd == NULL)
 			return (CMD_RETURN_NORMAL);
 		if (cmd_string_parse(cmd, &cmdlist, NULL, 0, &cause) != 0) {
